spec/rspec/mocks/any_instance_spec.rb

```diff
@@ -123,6 +123,11 @@ def private_method; :private_method_return_value; end
               expect(klass.new.one.plus(1)).to eq(2)
               expect(klass.new.one.plus(2)).to eq(3)
             end
+
+            it 'can use a chain of methods to perform an expectation' do
+              expect_any_instance_of(klass).to receive_message_chain('message1.message2').with('some args')
```

Comment:
I didn't realize we supported `with` with message chains.  Seems weird, since `with` is specifying something about a particular message but message chains are about multiple messages...

That said, I can see that works for normal doubles.